### PR TITLE
Build CFO image directly in OpenShift CI as part of CI run

### PR DESCRIPTION
**Pull request by ChristianZaccaria**
Related issue: https://github.com/project-codeflare/codeflare-operator/issues/359 

## Changes
- Building CFO image from Dockerfile provided.
- Planning to patch env variable in ODH once release is available.

## Thoughts
- Are we patching the env var from CFO repo for now? Will need to patch before building I think.
- `basic-dsc.yaml` should be copied over from distributed-workloads repo to codeflare-operator repo for it to be used.

### CODE REVIEW

File: ci-operator/config/opendatahub-io/codeflare-operator/opendatahub-io-codeflare-operator-main.yaml [hunk 1]
```diff
@@ -48,19 +51,20 @@ tests:
...
-          sed -i "s|<DW PR tarball URI>|https://api.github.com/repos/opendatahub-io/distributed-workloads/tarball/pull/${PULL_NUMBER}/head|" ./contrib/configuration/basic-dsc.yaml
+          sed -i "s|<CFO PR tarball URI>|https://api.github.com/repos/opendatahub-io/codeflare-operator/tarball/pull/${PULL_NUMBER}/head|" ./contrib/configuration/basic-dsc.yaml
         fi
         oc apply -f ./contrib/configuration/basic-dsc.yaml -n opendatahub
         cat ./contrib/configuration/basic-dsc.yaml
         sleep 60
         cd tests && go test -timeout 60m -v ./integration
+      dependencies:
+      - env: ODH-CODEFLARE-OPERATOR-CONTROLLER-IMAGE
+        name: image-registry.openshift-image-registry.svc:5000/openshift-operators/codeflare-operator
```
Comment (sutaakar) [inline ci-operator/config/opendatahub-io/codeflare-operator/opendatahub-io-codeflare-operator-main.yaml]:
```suggestion
        name: codeflare-operator
```

Comment (ChristianZaccaria) [reply ci-operator/config/opendatahub-io/codeflare-operator/opendatahub-io-codeflare-operator-main.yaml]: Yeah I just realized that, good catch thanks!

**Comment by anishasthana** [comment]
What's the ratoinale behind this guy?

**Comment by ChristianZaccaria** [comment]
> What's the ratoinale behind this guy?

This is part of resolving this issue: https://github.com/project-codeflare/codeflare-operator/issues/359 

(I should've tagged it)

**Comment by anishasthana** [comment]
Ah, I see -- we will be building a new image as part of the openshift-ci test run. The PR subject made me think we will use openshift-ci for image builds

**Comment by ChristianZaccaria** [comment]
> FYI the DSC CR should look like this (for using main branch, to test specific PR it needs to be adjusted ):
> 
> ```
> apiVersion: datasciencecluster.opendatahub.io/v1
> kind: DataScienceCluster
> metadata:
>   labels:
>     app.kubernetes.io/created-by: opendatahub-operator
>     app.kubernetes.io/instance: default
>     app.kubernetes.io/managed-by: kustomize
>     app.kubernetes.io/name: datasciencecluster
>     app.kubernetes.io/part-of: opendatahub-operator
>   name: example-dsc
> spec:
>   components:
>     codeflare:
>       devFlags:
>         manifests:
>           - uri: 'https://api.github.com/repos/opendatahub-io/codeflare-operator/tarball/main'
>             contextDir: 'config'
>             sourcePath: 'manifests'
>       managementState: Managed
>     dashboard:
>       managementState: Managed
>     datasciencepipelines:
>       managementState: Removed
>     kserve:
>       managementState: Removed
>     modelmeshserving:
>       managementState: Removed
>     ray:
>       devFlags:
>         manifests:
>           - uri: 'https://api.github.com/repos/opendatahub-io/kuberay/tarball/master'
>             contextDir: 'ray-operator/config'
>             sourcePath: 'default'
>       managementState: Managed
>     workbenches:
>       managementState: Managed
> ```
> 
> Edit: Updated CR

That makes perfect sense, thanks Karel! Adjusted that now. Unsure if it's the best way of doing it but I confirmed the sed commands I used for replacing those fields worked locally.

**Comment by sutaakar** [comment]
@ChristianZaccaria Can you adjust `oc logs -n openshift-operators deployment/codeflare-operator-manager > ${ARTIFACT_DIR}/codeflare-operator.log` to use `opendatahub` namespace? That should help figuring out what is the issue.

**Review by sutaakar — APPROVED** [review]
/lgtm

**Pull request merged**